CMakeLists.txt

```diff
@@ -624,14 +622,24 @@ EXECUTE_PROCESS(
     OUTPUT_STRIP_TRAILING_WHITESPACE)
 MESSAGE(STATUS "extracted version numbers from ${CMAKE_CURRENT_SOURCE_DIR}/include/h2o/version.h:\n\tprerelease=${IS_PRERELEASE}\n\tlibrary version=${LIBRARY_VERSION}\n\tlibrary soversion=${LIBRARY_SOVERSION}")
 
-SET_TARGET_PROPERTIES(libh2o PROPERTIES
-    OUTPUT_NAME h2o
-    VERSION ${LIBRARY_VERSION}
-    SOVERSION ${LIBRARY_SOVERSION})
+IF (LIBUV_FOUND)
+    SET_TARGET_PROPERTIES(libh2o PROPERTIES
+        OUTPUT_NAME h2o
+        VERSION ${LIBRARY_VERSION}
+        SOVERSION ${LIBRARY_SOVERSION})
+    TARGET_LINK_LIBRARIES(libh2o ${LIBUV_LIBRARIES})
+ELSE ()
+    SET_TARGET_PROPERTIES(libh2o PROPERTIES
+        OUTPUT_NAME h2o
+        COMPILE_FLAGS "-DH2O_USE_LIBUV=0"
+        VERSION ${LIBRARY_VERSION}
+        SOVERSION ${LIBRARY_SOVERSION})
+ENDIF (LIBUV_FOUND)
```

Comment:
FWIW, this change would be a problem for us, I do not think we want to build one library with different set of APIs, depending on if libuv exists.

`libh2o-evloop` is the library that we provide with `-DH2O_USE_LIBUV=0` set. This variant uses evloop (our own event loop) as the backend.

Reply:
Here I have added if else, if `LIBUV_FOUND` is set, then it compiles without compile flags and links to libuv, if it is unset, it does else condition where it skips linking to libuv and compile with `-DH2O_USE_LIBUV=0` flag, otherwise make breaks with `uv.h` not found errors